### ISSUE TITLE
pppd keeps running after disconnect

### ISSUE THREAD

**Issue by fnordahl** [nm-l2tp/network-manager-l2tp#39]
I am running on Ubuntu 16.04 LTS.

After disconnecting a L2TP/IPSec VPN connection using the NetworkManager Gnome applet in the top menu bar, I can see that the pppd process is not being killed and I have to kill it manually.

**Comment by dkosovic** [comment]
Just pppd or both pppd and its parent xl2tpd process ?

**Comment by fnordahl** [comment]
Just pppd.
```
$ ps axuw | egrep "pppd|xl2tpd"
root     18553  0.0  0.0 209520  6308 ?        Sl   07:43   0:00 /usr/sbin/pppd passive nodetach : file /var/run/nm-ppp-options.xl2tpd.18465 plugin pppol2tp.so pppol2tp 7
```

**Comment by dkosovic** [comment]
I'm pretty sure it is related to https://github.com/nm-l2tp/network-manager-l2tp/issues/38#issuecomment-288434451 where the Ubuntu modified version of xl2tpd crashes before it has had a chance to reap its child pppd process.

**Comment by fnordahl** [comment]
Ah, of course. I think you are right. I have observed xl2tpd die with SEGFAULT.

I will make an attempt at debugging the crash tomorrow.

**Comment by fnordahl** [comment]
https://bugs.launchpad.net/ubuntu/+source/xl2tpd/+bug/1677990

**Comment by fnordahl** [comment]
Please do help confirm the bug in launchpad by clicking "this bug affects me" @dkosovic @mgedmin

**Comment by fnordahl** [comment]
This should now be fixed in Ubuntu Xenial 16.04 LTS

**Comment by dkosovic** [comment]
That's awesome news! I'll update the xenial and yakkety network-manager-l2tpd PPD packages by putting explicit dependencies for xl2tpd >= these updated revisions.

I've uploaded a Debian Sid package to :
* https://mentors.debian.net/package/network-manager-l2tp

and requested sponsorship on the pkg-utopia-maintainers@lists.alioth.debian.org mailing list (as Utopia package the similar network-manager-pptp package). Hopefully I'll find a sponsor, once it is in Debian, I'll put a sync request in for Ubuntu and mention the xenial and yakkety specific depedencies for xl2tpd and strongswan.
